p2p/src/network.rs

```diff
@@ -276,7 +265,37 @@ impl<T: Pload, K: Kex, E: Enc> NetworkBase<T, K, E> {
         self.terminated_receivers.push(terminated_receiver);
     }
 
-    fn connect_peer(&mut self, ConnectPeer { peer_id: peer }: ConnectPeer) {
+    fn set_current_topology(&mut self, UpdateTopology(topology): UpdateTopology) {
+        iroha_logger::debug!(?topology, "Network receive new topology");
+        self.current_topology = topology;
+        self.update_topology()
+    }
+
+    fn update_topology(&mut self) {
+        let to_connect = self.current_topology
+            .iter()
+            // Peer is not connected but should
+            .filter(|peer| !self.peers.contains_key(&peer.public_key))
+            .cloned()
+            .collect::<Vec<_>>();
+
+        let to_disconnect = self.peers
+            .keys()
+            // Peer is connected but shouldn't
+            .filter(|public_key| !self.current_topology.contains(*public_key))
+            .cloned()
+            .collect::<Vec<_>>();
```

Comment:
Can this be done with some kind of set manipulation? For example substraction?

Reply:
Yes, it's doable, but will require slight changes in way how we store peers. I will do this.

Reply:
> And if you look how difference is implemented nothing fancy here it's almost equal to how i implemented it here.

Well it's mostly not about performance it's about readabillity and understanding what author meant.

But I see the point, that it doesn't worth it. Closing this